Utils.py

```diff
@@ -177,7 +180,7 @@ def user_path(*path: str) -> str:
     """Returns either local_path or home_path based on write permissions."""
     if hasattr(user_path, "cached_path"):
         pass
-    elif os.access(local_path(), os.W_OK):
+    elif os.access(local_path(), os.W_OK) and not (is_frozen() and is_macos):
```

Comment:
I think the two new conditions should be swapped
```suggestion
    elif os.access(local_path(), os.W_OK) and not (is_macos and is_frozen()):
```
imo it's easier to read (i.e. see that it's only relevant for macos)
and the short circuit is also faster (it's faster to check a variable than a function(); and for end-users is_macos is False in most cases while is_frozen() is True in most cases)